add docs for user-manual for volto frontend (#4027)
plone/volto: docs/source/user-manual/blocks.md

---
myst:
  html_meta:
    "description": "This is the end-User documentation for Plone 6 Volto frontend."

>     "property=og:description": "Volto is the React-based frontend for the Plone CMS. It is the default UI for the Plone 6 release."

Author: Ohh thanks this was something new to me.

    "property=og:title": "Frontend"

>     "keywords": "Volto, Plone, frontend, React, User manual"

Reviewer: ```suggestion
    "keywords": "Volto, Plone, frontend, React, User manual, edit blocks"
```

---

```{todo}
This page needs content
```

# Blocks

> Volto features the Pastanaga Editor Engine, allowing you to visually compose a page using blocks. The editor allows you to add, modify, reorder and delete blocks given your requirements. Blocks provide the user the ability to display content in an specific way, although they can also define behavior and have specific features.

Author: ohh I heard about 79 column rule but "Docs are not code" mantra is new to me 😅😂

> ## Types of pre-built Blocks

Reviewer: ```suggestion
## Choose a pre-built block type
```

> You can access and choose one block out of different types of pre-built blocks by:

Reviewer: ````suggestion
```{todo}
Add description of a Volto block.
```

Volto offers several default block types out of the box.
You can access and choose a block type to add to your content type when you have an empty block in it.

To create an empty block after an existing block, click in the block, then hit the {kbd}`Enter` key.
A new empty block appears.

```{todo}
Add screenshots or video illustrating the steps.
```

```{note}
There is a new experimental feature that places a `+` below every block in a content type.

See https://github.com/plone/volto/pull/3815 for details of the feature and how to enable it.
```

Now with your empty block available, you can select its type in one of two ways.
````

> 1. clicking on the `+` button provide on the left-side of empty block.

Reviewer: ```suggestion
1.  Click the `+` button on the left-hand side of the empty block.
```

Reviewer: Use this screenshot, nested within the list item.

![Screen Shot 2022-12-03 at 2 50 47 PM](https://user-images.githubusercontent.com/102112/205465377-d7dd7c69-bdbc-4702-baee-6ef0aa8102a1.png)

> 2. typing `/` inside empty block will open blocks menu.

Reviewer: ```suggestion
2. Type `/` inside empty block to open the blocks menu.
    ![Screen Shot 2022-12-03 at 2 52 11 PM](https://user-images.githubusercontent.com/102112/205465439-842aac3a-5fbc-4bb3-a4db-6ff687068467.png)
```

Use this screenshot, nested within the list item.

![Screen Shot 2022-12-03 at 2 52 11 PM](https://user-images.githubusercontent.com/102112/205465439-842aac3a-5fbc-4bb3-a4db-6ff687068467.png)

(user-manual-description-block-label)=

## Description Block

> Description block provides you blank textarea to write plain text. 

Reviewer: ```suggestion
A description block provides a blank textarea to write plain text.
```
